tests/unit/h/security/policy/_cookie_test.py

```diff
@@ -1,41 +1,137 @@
 from h.security.policy.helpers import AuthTicketCookieHelper
 
 
 class TestCookiePolicy:
-    def test_identity(self, cookie_policy, helper, pyramid_request):
+    def test_identity(self, cookie_policy, helper, pyramid_request, html_authcookie):
         identity = cookie_policy.identity(pyramid_request)
 
         helper.add_vary_by_cookie.assert_called_once_with(pyramid_request)
-        helper.identity.assert_called_once_with(sentinel.cookie, pyramid_request)
+        helper.identity.assert_called_once_with(html_authcookie, pyramid_request)
         assert identity == helper.identity.return_value
 
+    def test_identity_issues_api_authcookie(
+        self, cookie_policy, helper, pyramid_request, api_authcookie
+    ):
+        helper.remember.return_value = [
+            ("header_name_1", "header_value_1"),
+            ("header_name_2", "header_value_2"),
+        ]
+        del pyramid_request.cookies[api_authcookie.cookie_name]
+
+        cookie_policy.identity(pyramid_request)
+
+        pyramid_request._process_response_callbacks(  # pylint:disable=protected-access
+            pyramid_request.response
+        )
+        helper.remember.assert_called_once_with(
+            api_authcookie, pyramid_request, helper.identity.return_value.user.userid
+        )
+        for header in helper.remember.return_value:
+            assert header in pyramid_request.response.headerlist
+
+    @pytest.mark.parametrize(
+        "identity",
+        [
+            None,
+            Identity_(user=None, auth_client=None),
+        ],
+    )
+    def test_identity_doesnt_issue_api_authcookie_if_user_not_authenticated(
+        self,
+        cookie_policy,
+        helper,
+        pyramid_request,
+        api_authcookie,
+        identity,
+        assert_api_authcookie_not_issued,
+    ):
+        helper.identity.return_value = identity
+        del pyramid_request.cookies[api_authcookie.cookie_name]
+
+        cookie_policy.identity(pyramid_request)
+
+        assert_api_authcookie_not_issued()
+
+    def test_identity_doesnt_issue_api_authcookie_if_already_issued(
+        self, cookie_policy, pyramid_request, assert_api_authcookie_not_issued
+    ):
+        cookie_policy.identity(pyramid_request)
+
+        assert_api_authcookie_not_issued()
+
+    def test_identity_issues_api_authcookie_only_once(
+        self,
+        cookie_policy,
+        helper,
+        pyramid_request,
+        api_authcookie,
+        assert_api_authcookie_not_issued,
+    ):
+        helper.remember.return_value = [("header_name_1", "header_value_1")]
+        del pyramid_request.cookies[api_authcookie.cookie_name]
+        cookie_policy.identity(pyramid_request)
+        pyramid_request._process_response_callbacks(  # pylint:disable=protected-access
+            pyramid_request.response
+        )
+
+        cookie_policy.identity(pyramid_request)
+
+        helper.reset_mock()
+        assert_api_authcookie_not_issued()
```

Comment:
Bunch of slightly awkward tests for the transitional API auth cookie issuing. We'll be able to delete these after the 30-days transition period.